components/brave_mobile_subscription/renderer/android/subscription_render_frame_observer.cc

```diff
@@ -0,0 +1,201 @@
+#include "build/build_config.h"
+#include "content/public/renderer/render_frame.h"
+#include "third_party/blink/public/common/browser_interface_broker_proxy.h"
+#include "third_party/blink/public/platform/web_security_origin.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "url/url_util.h"
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+#include "brave/components/brave_vpn/common/brave_vpn_utils.h"
+#endif
+
+#if BUILDFLAG(ENABLE_AI_CHAT)
+#include "brave/components/ai_chat/core/common/features.h"
+#endif
+
+namespace brave_subscription {
+
+namespace {
+
+inline constexpr char kIntentParamName[] = "intent";
+inline constexpr char kIntentParamValue[] = "connect-receipt";
+inline constexpr char kIntentParamTestValue[] = "connect-receipt-test";
+inline constexpr char kProductParamName[] = "product";
+inline constexpr char kProductVPNParamValue[] = "vpn";
+inline constexpr char kProductLeoParamValue[] = "leo";
+
+}  // namespace
+
+SubscriptionRenderFrameObserver::SubscriptionRenderFrameObserver(
+    content::RenderFrame* render_frame,
+    int32_t world_id)
+    : RenderFrameObserver(render_frame), world_id_(world_id) {}
+
+SubscriptionRenderFrameObserver::~SubscriptionRenderFrameObserver() = default;
+
+bool SubscriptionRenderFrameObserver::EnsureConnected() {
+  bool bound = false;
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+  if (!vpn_service_.is_bound()) {
+    render_frame()->GetBrowserInterfaceBroker()->GetInterface(
+        vpn_service_.BindNewPipeAndPassReceiver());
+  }
+  bound |= vpn_service_.is_bound();
+#endif
+#if BUILDFLAG(ENABLE_AI_CHAT)
+  if (ai_chat::features::IsAIChatHistoryEnabled()) {
+    if (!ai_chat_subscription_.is_bound()) {
+      render_frame()->GetBrowserInterfaceBroker()->GetInterface(
+          ai_chat_subscription_.BindNewPipeAndPassReceiver());
+    }
+    bound |= ai_chat_subscription_.is_bound();
+  }
+#endif
+  return bound;
+}
+
+void SubscriptionRenderFrameObserver::DidCreateScriptContext(
+    v8::Local<v8::Context> context,
+    int32_t world_id) {
+  if (!render_frame()->IsMainFrame() || world_id_ != world_id) {
+    return;
+  }
+
+  if (!IsAllowed()) {
+    return;
+  }
+
+  auto connected = EnsureConnected();
+  if (!connected) {
+    LOG(ERROR) << "Failed to establish connection to a mojo channel";
+    return;
+  }
+
+  switch (product_) {
+    case Product::kVPN:
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+      if (vpn_service_.is_bound()) {
+        vpn_service_->GetPurchaseToken(
+            base::BindOnce(&SubscriptionRenderFrameObserver::OnGetPurchaseToken,
+                           weak_factory_.GetWeakPtr()));
+      }
+#endif
+      break;
+    case Product::kLeo:
+#if BUILDFLAG(ENABLE_AI_CHAT)
+      if (ai_chat_subscription_.is_bound()) {
+        ai_chat_subscription_->GetPurchaseToken(
+            base::BindOnce(&SubscriptionRenderFrameObserver::OnGetPurchaseToken,
+                           weak_factory_.GetWeakPtr()));
+      }
+#endif
+      break;
+    default:
+      NOTREACHED();
```

Comment:
when you remove `kNone`, these `default:` cases should also be removed.

Reply:
done